client/cli_main.py

```diff
@@ -306,35 +307,38 @@ def main():
                             else:
                                 while reading:
                                     print("Please choose a thought to read by entering its number or type B to go back:\n")
-                                    for i, thought in enumerate(thoughts):
-                                        print(f"{i+1}. TITLE: {thought['title']}\t RATING: {thought['rating']}")
-                                    
+                                    i = 0
+                                    for thought in thoughts:
+                                        print(f"{i + 1}. TITLE: {thought['title']}\t RATING: {thought['rating']}")
+                                        i += 1
                                     try:
                                         thought_choice = input("\nEnter thought number: ")
                                         if thought_choice == "b" or thought_choice == "B":
                                             reading = False
-                                            break
-                                        else: 
+                                        else:
                                             thought_num = int(thought_choice)
-                                            selected_thought = thoughts[thought_num-1]
+                                            selected_thought = thoughts[thought_num - 1]
                                             print(f"\nTITLE:  {selected_thought['title']}")
                                             print(f"RATING: {selected_thought['rating']}\n")
-                                            try:
-                                                decrypted_message = decrypt_message(selected_thought["content"].encode("utf-8"), encrypted_sym_key)
-                                                print(f"MESSAGE:  { decrypted_message}\n\n")
-                                                update_rating_for_thought(server_url, str(selected_thought["key"]))
-                                            except FileNotFoundError as err:
-                                                print("Error decrypting message, you may need to generate your keys still!\nError:", err)
-                                            except ValueError as err:
-                                                print("Please restart the programme to register your keys!\nError:", err)
+                                            decrypted_message = decrypt_message(selected_thought['content'], encrypted_sym_key)
+                                            print(f"MESSAGE:  {decrypted_message}\n\n")
+                                            key_path = os.path.join(os.path.dirname(__file__), 'keys', 'message.key')
+                                            with open(key_path, 'rb') as key_file:
+                                                key = key_file.readline(1)
+                                            update_rating_for_thought(server_url, key)
```

Comment:
I'm not sure the key is supposed to be a str type, so I changed it to bytes, but it still is throwing a 404 error...

Reply:
it is a string type normally, see my previous answer :-)